lib/stronger_parameters/controller_support/permitted_parameters.rb

```diff
@@ -0,0 +1,169 @@
+require 'stronger_parameters/constraints'
+
+module StrongerParameters
+  module ControllerSupport
+    module PermittedParameters
+      def self.included(klass)
+        klass.extend ClassMethods
+        klass.before_action :permit_parameters
+      end
+
+      class PermittedParametersHash < HashWithIndifferentAccess
```

Comment:
overriding has is never a good idea since users can use whatever methods they want and it will destroy the internal state ... better use a wrapper class ... but that's for another PR ...